app/scripts/modules/amazon/serverGroup/configure/serverGroupConfiguration.service.js

```diff
@@ -160,7 +160,10 @@ module.exports = angular.module('spinnaker.aws.serverGroup.configure.service', [
           .valueOf();
         if (command.keyPair && filtered.indexOf(command.keyPair) === -1) {
           var acct = command.backingData.credentialsKeyedByAccount[command.credentials] || {regions: [], defaultKeyPair: null};
-          command.keyPair = acct.defaultKeyPair;
+
+          // {{region}} is the only supported substitution pattern
+          command.keyPair = acct.defaultKeyPair.replace('{{region}}', command.region);
```

Comment:
Should wrap this in a conditional expression to avoid NPEs when no `defaultKeyPair` is configured for an account:
```javascript
if (acct.defaultKeyPair) {
  command.keyPair = acct.defaultKeyPair.replace('{{region}}', command.region);
}
```